### ISSUE TITLE
Unable to create 2 pull requests from the same branch

### ISSUE THREAD

**Issue by henninghall** [peter-evans/create-pull-request#1264]
#### Attempt
Trying to create 2 PRs from the same branch

#### Result
One PR is created and instantly closed. 
(when the second PR is about to be created, the first one gets closed instead)

#### Expected
Two different PRs to be created

### Steps to reproduce

1. Add these 2 steps to a workflow.

```yml
      - id: create_pr_to_development
        name: Create PR to development
        uses: peter-evans/create-pull-request@v4
        with:
          base: development
          branch: BRANCH_NAME
          delete-branch: false
          title: Merge `${{ env.branch_name }}` into `development`
          body: Automated PR from create-candidate github action. Merge this when candidate is released.

      - id: create_pr_to_main
        name: Create PR to main
        uses: peter-evans/create-pull-request@v4
        with:
          base: main
          branch: BRANCH_NAME
          delete-branch: false
          title: Merge `${{ env.branch_name }}` into `main`
          body: Automated PR from create-candidate github action. If you merge this, a new production build will be triggered.

```

2. Replace BRANCH_NAME with a branch that exists
3. Make sure branches `main` and `development` exists or rename those branches above 👆

**Comment by henninghall** [comment]
If someone has the same issue i managed to solve it by using `repo-sync/pull-request`

```yml
      - id: create_pr_to_development
        name: Create PR to development
        uses: repo-sync/pull-request@v2
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          destination_branch: "development"
          source_branch: ${{ env.branch_name }}
          pr_title: "Merge ${{ env.branch_name }} into development"
          pr_body: Automated PR from create-candidate github action. Merge this when candidate is released.

      - id: create_pr_to_main
        name: Create PR to main
        uses: repo-sync/pull-request@v2
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          destination_branch: "main"
          source_branch: ${{ env.branch_name }}
          pr_title: "Merge ${{ env.branch_name }} into main"
          pr_body: Automated PR from create-candidate github action. If you merge this, a new production build will be triggered.

```

**Comment by peter-evans** [comment]
Hi @henninghall 

You can create two PRs from the same source branch with this action. You just need to make sure that the PR branch (the `branch` input) is different for each PR.

**Comment by oliversalzburg** [comment]
@peter-evans I'm not sure that is correct, because I just ran into the same issue and I have two different `branch` configurations.

When `create-pull-request` wants to create the second PR, it will print `Resetting working base branch 'feat/other-pr'`, which is the `branch` from the previous PR and not mentioned in the second one. 

For illustration, my config looks something like this:
```yaml
- uses: peter-evans/create-pull-request@v4
  with:
    add-paths: |
      ./packages/payment-models/index.d.mts
    base: next
    branch: feat/update-tsd
    
- uses: peter-evans/create-pull-request@v4
  with:
    add-paths: |
      ./packages/infrastructure/lambda-schema.yaml
    base: next
    branch: feat/update-openapi-components
```

I assume that I would fix this behavior by setting the `base` explicitly, but that had no impact on the behavior.

**Comment by peter-evans** [comment]
@oliversalzburg I can see what you are trying to do, but unfortunately the action doesn't support executing the action twice in a row with different `add-paths`. This is because any other updates are reset during creating the first PR. It an interesting use case and it's made me realise that it might be good to try and support it. When I have some time I will try and experiment with restoring other changes that exist locally.

In the meantime, please take a look at the following answer that has some examples of how to tweak your workflow to create multiple PRs.
https://github.com/peter-evans/create-pull-request/issues/1302#issuecomment-1296404162

**Comment by peter-evans** [comment]
@oliversalzburg I've worked on an enhancement to the action to support your use case. It's a potentially breaking behaviour change for the action, so I will need to release it as a new major version. Until the release, you can use the beta version with tag `v5-beta`.

You should be able to do this in your workflow now:
```yml
- uses: peter-evans/create-pull-request@v5-beta
  with:
    add-paths: |
      ./packages/payment-models/index.d.mts
    branch: feat/update-tsd
    
- uses: peter-evans/create-pull-request@v5-beta
  with:
    add-paths: |
      ./packages/infrastructure/lambda-schema.yaml
    branch: feat/update-openapi-components
```
(I removed setting the `base` explicitly because it should be unnecessary if you are already checked out on the `next` branch when the action executes)

**Comment by oliversalzburg** [comment]
Oh nice! Seems like I missed the earlier notification.

Right now, we're using multiple workflows, and already build a lot of stuff around that approach now. The pipelines I was working on earlier, served as blueprints for another team who is now in charge of handling all the CI.

As I am using your action in a lot of places, I am very likely to get back to this though. I very much appreciate you looking into this and I will share my feedback once I have it :)

**Comment by peter-evans** [comment]
This is now fixed in `v5.0.0` / `v5`.

If you are using `v5-beta` or `v5-rc`, please update to `v5`. I will remove pre-release versions after 4 weeks. Thank you for testing the pre-release version!
